pulsar-client/src/main/java/org/apache/pulsar/client/impl/BatchMessageIdImpl.java

```diff
@@ -147,6 +147,12 @@ public MessageIdImpl prevBatchMessageId() {
             ledgerId, entryId - 1, partitionIndex);
     }
 
+    // MessageIdImpl is widely used as the key of a hash map, in this case, we should convert the batch message id to
+    // have the correct hash code.
+    public MessageIdImpl getMessageIdImpl() {
```

Comment:
I would name this 'toMessageIdImpl()'

Reply:
Good suggestion. `get` usually means returning an internal field (`A` -> `A.x`), while `to` usually means the conversion (`A -> B`). Addressed.